### ISSUE TITLE
Backward compatibility of java code is breaking when upgrading the api from v5.42 to v6.x

### ISSUE THREAD

**Issue by ashish777gupta** [microsoftgraph/msgraph-sdk-java#1980]
While changing the graph api from v5.42 to v.6x the code is breaking and most of the classes are not found like, TokenCredentialAuthProvider, GraphServiceException, WorkbookWorksheetAddParameterSet.

**Code for creating the worksheet:**
`import com.microsoft.graph.http.GraphServiceException;
import com.microsoft.graph.models.*;
import com.microsoft.graph.options.HeaderOption;
import com.microsoft.graph.requests.GraphServiceClient;
import com.microsoft.graph.requests.WorkbookWorksheetRangeRequest;
private GraphServiceClient client;
public boolean createWorksheet(String worksheetName) {
        try {
            // API call to create new worksheet
            client.me()
                    .drive()
                    .root()
                    .itemWithPath(workbookPath)
                    .workbook()
                    .worksheets()
                    .add(
                            WorkbookWorksheetAddParameterSet.newBuilder()
                              .withName(Utils.encodeString(worksheetName))
                                    .build())
                    .buildRequest(workbookSessionIdHeader)
                    .post();
            setActiveSheetName(worksheetName);
            return true;
        } catch (GraphServiceException exception) {
            System.err.println("Graph service encountered an error: " + exception.getMessage());
        }
    }`

**Error:**
Cannot resolve method 'me()'
Cannot resolve symbol 'WorkbookWorksheetAddParameterSet'
Cannot resolve symbol 'GraphServiceException'

**Comment by Ndiritu** [comment]
@ashish777gupta thanks for reaching out.

Yes, there are expected breaking changes due to the major version bump. Please see the [upgrade guide](https://github.com/microsoftgraph/msgraph-sdk-java/blob/dev/docs/upgrade-to-v6.md) and feel free to reach out if anything is not clear.

**Comment by ashish777gupta** [comment]
I have reviewed the guide and found that most of the classes have been removed or changed in comparison to the older API. Consequently, is there any way to utilize previously written code with the newer API since almost all of the code triggers errors.

For example, the guide specifies that [TokenCredentialAuthProvider] has been removed. Instead, if custom authentication flow is needed, a class implementing [AccessTokenProvider] must be created alongside [BaseBearerTokenAuthenticationProvider], and then passed to the GraphServiceClient constructor. This process is required to obtain a custom token. Is there no possibility to utilize [TokenCredentialAuthProvider] in the latest version?

Also, in the previous version of the Graph API (v5.42), to create a worksheet, we used the following API call, which now produces errors in v6:
"Cannot resolve method 'root' in 'DriveRequestBuilder'" and "Cannot resolve symbol 'WorkbookWorksheetAddParameterSet'".
`client.me()
    .drive()
    .root()
    .itemWithPath(workbookPath)
    .workbook()
    .worksheets()
    .add(
        WorkbookWorksheetAddParameterSet.newBuilder()
            .withName(Utils.encodeString(worksheetName))
            .build())
    .buildRequest(workbookSessionIdHeader)
    .post();
`
